tests/topotests/bgp_vpnv4_per_nexthop_label/test_bgp_vpnv4_per_nexthop_label.py

```diff
@@ -0,0 +1,818 @@
+    for entry in group:
+        dump = router.vtysh_cmd("show bgp ipv4 vpn {} json".format(entry), isjson=True)
+        for rd in dump:
+            assert (
+                "prefix" in dump[rd].keys()
+            ), "{0}, {1}, route distinguisher not present".format(router.name, entry)
+            assert dump[rd]["prefix"] == str(
+                entry
+            ), "{0}, {1}, ipv4 vpn entry not present".format(router.name, entry)
+            assert (
+                "paths" in dump[rd].keys()
+            ), "{0}, {1}, paths keyword not present}".format(router.name, entry)
+            for path in dump[rd]["paths"]:
+                assert (
+                    "remoteLabel" in path.keys()
+                ), "{0}, {1}, remoteLabel not present".format(router.name, entry)
+                logger.info(
+                    "{0}, {1}, label value is {2}".format(
+                        router.name, entry, path["remoteLabel"]
+                    )
+                )
+                if not storedLabelInited:
+                    storedLabel = path["remoteLabel"]
+                    storedLabelInited = True
+                    if isinstance(labelList, set):
+                        assert (
+                            storedLabel not in labelList
+                        ), "{0}, {1}, label already detected in a previous entry".format(
+                            router.name, entry
+                        )
+                        labelList.add(storedLabel)
+                else:
+                    assert (
+                        path["remoteLabel"] == storedLabel
+                    ), "{0}, {1}, label value not expected one (expected {2}, observed {3}".format(
+                        router.name, entry, storedLabel, path["remoteLabel"]
+                    )
+                if label_value_expected:
+                    assert (
+                        path["remoteLabel"] == label_value_expected
+                    ), "{0}, {1}, label value not expected (expected {2}, observed {3}".format(
+                        router.name, entry, label_value_expected, path["remoteLabel"]
+                    )
+
+
+def bgp_vpnv4_table_check_all(router, labelList=None, same=False):
+    """
+    Dump and check that vpnv4 entries are correctly configured with specific label values
+    * 'router': the router to check
+    * 'labelList': check that the label values are not present in the vpnv4 entries
+    *              that list is updated with the present label value found.
+    * 'same': by default, set to False. Addresses groups are classified by addresses.
+    *         if set to True, all entries of all groups should have a unique label value
+    """
+    group1 = ["172.31.0.11/32", "172.31.0.20/32", "172.31.0.111/32"]
+    group2 = ["172.31.0.12/32", "172.31.0.15/32"]
+    group3 = ["172.31.0.13/32"]
+    group4 = ["172.31.0.14/32"]
+    group5 = ["192.168.255.0/24", "192.0.2.0/24"]
+    if same:
```

Comment:
instead of groupX, prefixes_r11, prefixes_redist...

Reply:
done